pkg/network/iface/bond.go

```diff
@@ -165,11 +165,36 @@ func getSlaves(index int) ([]netlink.Link, error) {
 	return links, nil
 }
 
-func compareBond(b1, b2 *netlink.Bond) bool {
-	if b1.Name == b2.Name && b1.MTU == b2.MTU && b1.Mode == b2.Mode &&
-		b1.HardwareAddr.String() == b2.HardwareAddr.String() && b1.TxQLen == b2.TxQLen {
-		return true
+func compareBond(old, new *netlink.Bond) bool {
+	if old.Name != new.Name {
+		return false
 	}
 
-	return false
+	if old.Mode != new.Mode {
+		return false
+	}
+
+	// skip if mtu is omitted
+	if new.MTU != 0 && old.MTU != new.MTU {
+		return false
+	}
+
+	// skip if hardware addr is omitted
+	if new.HardwareAddr.String() != "" && old.HardwareAddr.String() != new.HardwareAddr.String() {
```

Comment:
Question:
If the new is "" and the old is not, is returning false what we expect?
Or is the new value the default, but the old value is not the default, and returning false is what we expect?

Reply:
- The compare function is just non-exported function and the new bond will never be nil in the current implementation.
- Yes, once the new value is the default, skip it and don't recreate the bond.